pkg/component/pushed_component.go

```diff
@@ -250,10 +250,13 @@ func getSource(component provider) (string, string, error) {
 }
 
 func getType(component provider) (string, error) {
-	if componentType, ok := component.GetLabels()[componentlabels.ComponentTypeLabel]; ok {
+	if componentType, ok := component.GetAnnotations()["odo.dev/project-type"]; ok {
 		return componentType, nil
+	} else if _, ok = component.GetLabels()[componentlabels.ComponentTypeLabel]; ok {
+		klog.V(1).Info("No annotation assigned; retuning 'Not available' since labels are assigned. Annotations will be assigned when user pushes again.")
+		return NOTAVAILABLE, nil
```

Comment:
Since we will be using Annotations to pass the actual component type, we use Annotations to retrieve it as well, but for an existing pushed component, this annotation is not added, it will be added when it is pushed again, until then, we can either return the component type retrieved from label or return `Not available`, which one should it be?

Reply:
@dharmit @kadel ^

Reply:
To support backwards compatibility, I would suggest retrieving it from the label for cases where it's not available in the annotations.